helpers/class.File.php

```diff
@@ -112,6 +112,28 @@ public static function isFileInsideDirectory($filename, $directory)
         if (false === $canonicalFilename) {
             return false;
         }
+
+        return 0 === strpos($canonicalFilename, $canonicalDirectory);
+    }
+
+    /**
+     * Helps prevent 'path traversal' attacks
+     *
+     * @param string $filename  absolute path to file
+     * @param string $directory absolute path to directory
+     * @return bool
+     */
+    public static function isAbsoluteFileInsideDirectory($filename, $directory)
```

Comment:
What if we use the previous method `isFileInsideDirectory` to validate  file despite if is absolute path or not?
I think a check like 
```
canonicalFilename = realpath( $filename );

if ( canonicalFilename !== $filename ) {
         $canonicalFilename = realpath($canonicalDirectory . DIRECTORY_SEPARATOR . $filename);
    }
```
What do you think?

Reply:
The reason why I added this is because originally, the tested file structure was this one:
```
path
`-- to
    `-- uncompressed package.zip
        |-- item1
        |   `-- item1.xml
        `-- shared
            `-- shared stimulus.xml
```
with the code in `item1.xml` was including `../shared/shared-stimulus.xml`, and the test was actually checking if `/path/to/uncompressed package.zip/item1/../shared/shared-stimulus.xml` was actually inside '/path/to/uncompressed package.zip/item1/', which is obviously not the case.
The idea was to prevent exposing the whole filesystem to the user.

What I did was to check if the absolute `/path/to/uncompressed package.zip/shared/shared stimulus.xml` is actually inside `/path/to/uncompressed package.zip/`.